assets/hip-1056/protobuf/stream/trace/smart_contract_service.proto

```diff
@@ -0,0 +1,196 @@
+    }
+
+    /**
+    * The storage value in this slot, may be left-padded with zeros to 32 bytes.
+    */
+    bytes read_value = 3;
+}
+
+/**
+ * Information about the initcode of a contract whose creation was attempted
+ * during the EVM transaction.
+ */
+message ContractInitcode {
+    oneof code {
+        /**
+         * The initcode used for a failed top-level creation.
+         */
+        bytes failed_initcode = 2;
+
+        /**
+         * The initcode "bookends" around a newly created contract's runtime bytecode.
+         */
+        ExecutedInitcode executed_initcode = 3;
+    }
+}
+
+/**
+  * Information about the initcode executed to create a contract.
+ */
+message ExecutedInitcode {
+    /**
+     * The id of the contract created by the initcode.
+     */
+    proto.ContractID contract_id = 1;
+
+    oneof initcode {
+        /**
+         * If the runtime bytecode follows the common convention of appearing as a subsequence
+         * of the initcode, the initcode pieces that surround the runtime bytecode (which is
+         * necessarily in the Hiero transactional unit's state changes).
+         */
+        InitcodeBookends initcode_bookends = 2;
+
+        /**
+         * If the runtime bytecode is not a subsequence of the initcode, the explicit initcode.
+         */
+        bytes explicit_initcode = 3;
+    }
+}
+
+/**
+  * The initcode bookends of a contract, which are the deploy bytecode and metadata
+  * bytecode that surround the runtime bytecode in a contract's deployment.
+  * <p>
+  * The runtime bytecode is not included here, as it is externalized via a state change
+  * with the matching contract id in the Hiero transactional unit.
+ */
+message InitcodeBookends {
+    /**
+     * The bytecode that precedes the runtime bytecode of a contract in initcode.
+     */
+    bytes deploy_bytecode = 1;
+
+    /**
+     * The bytecode that follows the runtime bytecode of a contract in initcode.
+     */
+    bytes metadata_bytecode = 2;
+}
+
+/**
+ * A EVM transaction log; c.f. eth_getTransactionReceipt.<br/>
+ * Stream consumers may compute bloom values from topics and data if desired.
+ */
+message EvmTransactionLog {
+    /**
+     * The contract emitting the log.
+     */
+    proto.ContractID contract_id = 1;
+
+    /**
+     * The logged data.
+     */
+    bytes data = 2;
+
+    /**
+     * The log's topics; may be left-padded with zeros to form 256-bit words.
+     */
+    repeated bytes topics = 3;
+}
+
+/**
+ * EVM trace data, including:
+ * 1. Initcode used for any internal contract creations.
+ * 2. Contract actions executed during the transaction (i.e., the call trace)
+ * 3. Contract slot usages, including reads and writes.
+ * 4. Events logged during the transaction.
+ * 5. Full error message, if any, produced by the contract call.
+ */
+message EVMTraceData {
+    /**
+     * The initcode of child contracts created during execution.
+     */
+    repeated ContractInitcode initcodes = 1;
```

Comment:
is `repeated` necessary because only the top level transaction will have statechanges? If so, we need to make sure that the information in `EVMTraceData` either only exist for the top level transaction, or we have a way to map them to corresponding child transaction's `ContractFunctionResult` / `ContractStateChange`.